File: core/core-domain/src/main/java/com/coffee/gu/stamp/StampHistoryItem.java
package com.coffee.gu.stamp;

import com.coffee.gu.enums.StampHistoryType;
import com.coffee.gu.enums.StampState;

import java.time.LocalDateTime;

public record StampHistoryItem(
        StampHistoryType type,
        LocalDateTime displayDate,
        long quantity,
        String storeName,
        LocalDateTime expiredAt
) {

    public static StampHistoryItem from(StampHistory stampHistory) {
        return new StampHistoryItem(
                stampHistory.getType(),
                stampHistory.getRecordedAt(),
                stampHistory.getQuantity(),
                stampHistory.getStoreName(),
                stampHistory.getExpiredAt()
        );
    }
}

File: core/core-domain/src/main/kotlin/com/coffee/gu/stamp/StampHistoryItem.kt
package com.coffee.gu.stamp

import com.coffee.gu.enums.StampHistoryType
import java.time.LocalDateTime

class StampHistoryItem(
    val type: StampHistoryType,
    val displayDate: LocalDateTime,
    val quantity: Int,
    val storeName: String,
    val expiredAt: LocalDateTime?,
) {
    companion object {
        fun from(stampHistory: StampHistory): StampHistoryItem {
            return StampHistoryItem(
                type = stampHistory.type,
                displayDate = stampHistory.recordedAt,
                quantity = stampHistory.quantity,
                storeName = stampHistory.storeName,
                expiredAt = stampHistory.expiredAt,
            )
        }
    }
}
